### ISSUE TITLE
Mulle-Objc Directory

### ISSUE THREAD

**Issue by clozinski** [mulle-objc/mulle-objc.github.io#3]
I am a huge fan of Objective-C, so I created a Mulle-Objc directory. 
http://blogory.org/mulle-objc

I invite you to browse the links, and add any that I missed. 

Warm Regards
Christopher Lozinski

**Comment by mulle-nat** [comment]
Thanks for the support. Looks good to me.

**Comment by clozinski** [comment]
You are most welcome.   It is so important that the market have an fully open source tool chain for Objective-C!  Many years ago I chose not to port BPG BlOCS to QNX because I did not want to get infected by the GNU license.  

There are three questions that everyone will have. First is what is the license?
So I added a page.  http://blogory.org/mulle-objc-license Did I get it correct?

The next question is how does the LLVM tool chain compare to the GNU tool chain?  Particularly on ARM chips?  If you could add a link that would interest people a lot.  I am not enough of a technical expert in this area to know which articles to recommend. 

The third and most interesting question is how does LLVM on ARM compare to GNU on Apple's customized arm chips?    What optimizations have you and Apple done to the Objective-C messenger on those platforms?  I wonder if Apple added custom hardware to the ARM microprocessor to accelerate message sending?

My readers want to know.   For many years I published the OpenStep Newsletter.  The release of your product is very very interesting to my readers.  The number of people who want to follow your product, is way way higher than the number of users you have.  People can watch stuff on git, but then the average reader gets too much noise.  If you have news announcements, and are kind enough to post them on my website, http://blogory.org/mulle-objc, then subscribers will just get major announcements.
All my blogory.org/iphone/rss subscribers will also get your news announcements. 

In a few days I will send an email out to my ancient OpenStep Newsletter mailing list telling them about your product.  If you are going to add any content, to blogory.org let me know, and I will hold off until you have done so.  

Warm Regards
Chris

**Comment by mulle-nat** [comment]
Hi Chris

thank for your efforts making this little project more well known.

So here are the answers:

Q: What are the licenses ?
A: The compiler is LLVM licensed like the original, no change.
All the libraries have been released with BSD-3 license. So you can use all of it in commercial closed-source projects at no cost, but you need to add the licenses to your documentation.

Q: How does the LLVM tool chain compare to the GNU tool chain ?
A: I personally can't really give a good answer. There are just too many comparison points. Here are some recent [benchmarks](https://www.phoronix.com/scan.php?page=article&item=gcc-61-clang39&num=1). In some sense the LLVM compiler **is** the Apple compiler and the **mulle-objc** compiler. Code-completion as an example should just work identically. My feeling is that LLVM is currently, all things considered, technologically ahead.  I think even [RMS](https://gcc.gnu.org/ml/gcc/2014-01/msg00247.html) sees it that way. Will that still be the case after gcc7 is released ?

Q: How does LLVM on ARM compare to GNU on Apple's customized arm chips? 
A: I don't really know. I can just give you guesses. Last time I looked (it's been a while), I saw no special Apple ARM opcodes to facilitate method sends in the [Apple runtime](https://opensource.apple.com/tarballs/objc4/). ARM being RISCy it would be kinda strange anyway. The mulle-objc compiler should be able to output ARM code right now, so interested readers may peruse the disassembly output of **mulle-clang** and Apple **clang** and draw their conclusions.

I am happy to answer all of your quesions here, but I don't really want to maintain something offsite, my time being somewhat limited and there is not a drought of things to do at the moment.

**Comment by clozinski** [comment]
Thank you for the excellent answers. 

You can see the content you contributed here. 
http://blogory.org/private/236

Please read the LLVM vs Apple question,  and question on arm chips to make sure I got it right.  I wrongly thought Apple was still on GCC, so I rewrote the questions to make more sense to our readers.  Hopefully I did not make any technical mistakes, but I am not as close to this technology as you are. 
So do check it. 

If there is a problem, you can log in to edit.  I am  sending you credentials to your info@codeon.de address.  

Warm Regards
Chris

**Comment by mulle-nat** [comment]
Looks good, though the company name isn't quite that biblical. Codeon rather than Gideon :)

**Comment by clozinski** [comment]
Fixed

I posted it on Reddit a few hours ago.   You still have 8 up votes.  i only have my  own 1 upvote. 
It is not like the world is burning a path to your door.  The software world is so driven by fads.  I saw the same thing happen in 1994 when the whole world abandoned Objective-C for C++.  
 
I will email it out tomorrow.  

Actually I am looking for Objective-C developers.   Maybe we are just hard to find.  

Warm Regards
Chris
